packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts

```diff
@@ -1,47 +1,94 @@
-import {RecordValue, Schema, StringValue} from '@coveo/bueno';
-import {getOrganizationEndpoints} from '../../api/platform-client';
+import {BooleanValue, RecordValue, Schema, StringValue} from '@coveo/bueno';
 import {CartInitialState} from '../../controllers/commerce/context/cart/headless-cart';
 import {ContextOptions} from '../../controllers/commerce/context/headless-context';
 import {cartDefinition} from '../../features/commerce/context/cart/cart-validation';
 import {contextDefinition} from '../../features/commerce/context/context-validation';
-import {PlatformEnvironment} from '../../utils/url-utils';
 import {
+  nonEmptyString,
+  requiredNonEmptyString,
+} from '../../utils/validate-payload';
+import {
+  AnalyticsConfiguration,
   EngineConfiguration,
   engineConfigurationDefinitions,
 } from '../engine-configuration';
 
 /**
- * The commerce engine configuration options.
+ * The commerce engine configuration.
  */
 export interface CommerceEngineConfiguration extends EngineConfiguration {
+  /**
+   * The commerce analytics configuration.
+   */
+  analytics: Pick<
+    AnalyticsConfiguration,
+    'enabled' | 'proxyBaseUrl' | 'source' | 'trackingId'
```

Comment:
The other analytics configuration options are not relevant in the commerce use case, as only the next mode is available.